### PR TITLE
Change --gpus to override when set and handle different vllmarg styles

### PR DESCRIPTION
**Pull request by danmcp**
Previously, --gpus wouldn't take effect if --tensor-parallel-size was set in serve vllm_args.  The normal behavior is that options should override config.  And a user could very possibly want to pass a different value than they use for the serve config.  Whether that value is set as an option or as evaluate config.

Also, the previous logic wasn't handling the --tensor-parallel-size=value format

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/model/evaluate.py [hunk 1]
```diff
@@ -258,13 +258,18 @@ def launch_server(
...
-                eval_serve.vllm.vllm_args.extend(["--tensor-parallel-size", str(gpus)])
-            else:
-                click.echo(
-                    "Ignoring --gpus with --tensor-parallel-size configured in serve vllm_args"
+            # Warn when overriding from vllm_args
+            tps_prefix = "--tensor-parallel-size"
+            if any(
+                s == tps_prefix or s.startswith(tps_prefix + "=")
+                for s in eval_serve.vllm.vllm_args
+            ):
```
Comment (tiran) [inline src/instructlab/model/evaluate.py:266]: It's a common problem. Let's make this a helper function in `utils`. This should do the trick:

```python
def split_arg_options(args: list[str]) -> set[str]:
    return set(arg.split("=", 1)[0]) for arg in args of arg.startswith("--"))
```

Comment (danmcp) [reply src/instructlab/model/evaluate.py:266]: I was hoping to use contains_argument from https://github.com/instructlab/instructlab/pull/1636/files once it goes in.

Comment (leseb) [reply src/instructlab/model/evaluate.py:266]: @danmcp I think this one might go in before #1636

Comment (danmcp) [reply src/instructlab/model/evaluate.py:266]: > @danmcp I think this one might go in before #1636

If it does, I'll can do a follow-on.  Or I can create the function in utils if that's preferred.

**Review by nathan-weinberg — APPROVED** [review]

**Review by alinaryan — APPROVED** [review]

**Pull request merged**